### PR TITLE
feat: connect VASP list to API

### PR DESCRIPTION
**Pull request by elysee15**
Fixes SC-597

### CODE REVIEW

File: web/gds-admin-ui/src/utils/index.js [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+function defaultEndpointPrefix() {
+    if (config.GDS_API_URL) {
+        return config.GDS_API_URL
+    }
+
+    switch (process.env.NODE_ENV) {
+        case ENVIRONMENT.DEV:
+            return "http://localhost:4434/v2"
+        case ENVIRONMENT.PROD:
+            return "https://api.admin.vaspdirectory.net"
```
Comment (bbengfort) [inline web/gds-admin-ui/src/utils/index.js:16]: Should be `https://api.admin.vaspdirectory.net/v2` - but actually we'll need to switch on `IS_TESTNET` here. e.g.

```
case ENVIRONMENT.PROD:
    if (config.IS_TESTNET) {
      return "https://api.admin.trisatest.net/v2";
   } else {
     return "https://api.admin.vaspdirectory.net/v2";
   }
```

File: web/gds-admin-ui/src/utils/index.js [hunk 2]
```diff
@@ -1,6 +1,29 @@
+import config from '../config';
+import { ENVIRONMENT } from '../constants';
+
 export * from './array';
 
 
+function defaultEndpointPrefix() {
```
Comment (bbengfort) [inline web/gds-admin-ui/src/utils/index.js:7]: Could we also write some tests for the utils in this function? I'd like to make sure we aren't creating too much technical debt for testing; but we can get to the bulk of that next week or the week after.

File: web/gds-admin-ui/src/pages/app/lists/index.js [hunk 3]
```diff
@@ -128,11 +132,11 @@ const VaspsList = (): React$Element<React$FragmentType> => {
...
                             </Row>
                             {
-                                vasps && vasps.length &&
+                                data && data.vasps && data.vasps.length &&
                                 <Table
                                     columns={columns}
-                                    data={vasps}
-                                    pageSize={5}
+                                    data={data?.vasps}
+                                    pageSize={data.page_size || 100}
```
Comment (bbengfort) [inline web/gds-admin-ui/src/pages/app/lists/index.js:139]: So does the data table map to the page size; meaning as you paginate in the table you're making requests to the server? I guess I'm not familiar with how redux and the API are interacting here.

File: web/gds-admin-ui/src/constants/index.js [hunk 4]
```diff
@@ -1,2 +1,7 @@
 export * from './layout';
 export * from './dashboard'
+
+export const ENVIRONMENT = {
+    DEV: "development",
+    PROD: "production"
+}
```
Comment (bbengfort) [inline web/gds-admin-ui/src/constants/index.js:7]: Nice :-)

File: web/gds-admin-ui/src/config.js [hunk 5]
```diff
@@ -1,6 +1,8 @@
 const config = {
-    API_URL: process.env.REACT_APP_API_URL,
-    GOOGLE_CLIENT_ID: process.env.REACT_APP_GOOGLE_CLIENT_ID
+    ENVIRONMENT: process.env.NODE_ENV,
+    GOOGLE_CLIENT_ID: process.env.REACT_APP_GOOGLE_CLIENT_ID,
+    IS_TESNET: process.env.REACT_APP_GDS_IS_TESTNET,
+    GDS_API_URL: process.env.REACT_APP_GDS_API_ENDPOINT
 };
 
 export default config;
```
Comment (bbengfort) [inline web/gds-admin-ui/src/config.js:8]: Thank you for putting this together; I like what we're doing here!

File: web/gds-admin-ui/src/utils/index.js (hunk 2 above)
Comment (elysee15) [reply web/gds-admin-ui/src/utils/index.js:7]: alright

File: web/gds-admin-ui/src/pages/app/lists/index.js (hunk 3 above)
Comment (elysee15) [reply web/gds-admin-ui/src/pages/app/lists/index.js:139]: Yes, it does. For the page size I don't make request currently

Comment (elysee15) [reply web/gds-admin-ui/src/pages/app/lists/index.js:139]: I take all the data and paginate it on my side

Comment (elysee15) [reply web/gds-admin-ui/src/pages/app/lists/index.js:139]: But I'm going to fix it soon

File: web/gds-admin-ui/src/__tests__/utils.spec.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { defaultEndpointPrefix } from "../utils";
+
+describe("defaultEndpointPrefix", () => {
+
+
+    it("should return the correct dev url prefix", () => {
+        const devUrlPrefix = "http://localhost:4434/v2"
+
+        expect(defaultEndpointPrefix()).toBe(devUrlPrefix);
```
Comment (bbengfort) [inline web/gds-admin-ui/src/__tests__/utils.spec.js:9]: Perfect, thank you!

File: web/gds-admin-ui/src/utils/index.js [hunk 7]
```diff
@@ -1,6 +1,33 @@
...
+
+    switch (process.env.NODE_ENV) {
+        case ENVIRONMENT.DEV:
+            return "http://localhost:4434/v2"
+        case ENVIRONMENT.PROD:
+            if (config.IS_TESTNET) {
+                return "https://api.admin.trisatest.net/v2";
+            } else {
+                return "https://api.admin.vaspdirectory.net/v2";
+            }
```
Comment (bbengfort) [inline web/gds-admin-ui/src/utils/index.js:20]: Looks great!

**Review by bbengfort — APPROVED** [review]
I'm glad that was fairly straight forward!

**Pull request merged**